YAPA/MainWindow.xaml.cs

```diff
@@ -587,14 +589,21 @@ private void ResetTicking()
             _stopWatch.Reset();
             _dispacherTime.Stop();
             CurrentTimeValue = "00:00";
-            CurrentPeriod.Text = "";
             _period = 0;
             _isBreak = false;
             _isBreakLong = false;
             _isWork = true;
             ProgressValue = .0;
             ProgressState = "None";
             _ticks = 0;
+
+            var lastDay = _itemRepository.GetPomodoros().Last();
+            if (lastDay.DateTime == DateTime.Now.Date)
+            {
+                _period = lastDay.Count;
```

Comment:
_period can be from 1 to 4 (1,2,3 - has pause interval of 5min, 4 - interval has 15min pause), lastDay.Count will return amount of pomodoros complete per day which can be 10,12,20... and application would break on 578 line check. We should take % 4 to allow user continue with next interval.